Src/libCZI/stdAllocator.cpp

```diff
@@ -30,24 +30,27 @@ void* CHeapAllocator::Allocate(std::uint64_t size)
 	{
 		throw std::out_of_range("The requested size for allocation is out-of-range.");
 	}
-#if defined(__EMSCRIPTEN__)||defined(__APPLE__)
-	return malloc((size_t)size);
-#else
-#if defined(__GNUC__)
+	// it's not clear to me why the code needs the returned pointer to be at a memory boundary
+	// meaning divisible by 32 in this case. I've preserved that functionality but restructured
+	// the code block to be more general.
+	// Compilation was failing on the original with linux distros that don't have the _ISOC11_SOURCE.
+#if defined(_WIN32)
+	return (void *)_aligned_malloc((size_t)size, 32);
+#elif defined(_ISOC11_SOURCE)
 	return aligned_alloc(32, size);
 #else
-	void* pv = _aligned_malloc((size_t)size, 32);
-	return pv;
-#endif
+	return ::operator new((size_t)size);
```

Comment:
can you confirm that the use of operator new will give the 32bit alignment?

Reply:
it won't, this code is replacing a malloc but malloc is pure C not C++. 
This is functionally equivalent to malloc but if library enhancements have been made they'll be in new not in malloc.

Reply:
oh yes!  but malloc and new are not exactly equivalent. this is a subtle behavior change that I don't think you intend.